web/src/message_view.ts

```diff
@@ -181,6 +181,8 @@ function create_and_update_message_list(
         if (!filter.requires_adjustment_for_moved_with_target) {
             const superset_datasets = message_list_data_cache.get_superset_datasets(filter);
             for (const superset_data of superset_datasets) {
+                // Reset properties that might have been set.
+                reset_id_info(id_info);
```

Comment:
Hmm, I'm not sure this is correct -- don't we need to re-initialize from `opts.id_info`?

Reply:
Oops, changed it.

```diff
+        const original_id_info = {...id_info};
         // Populate the message list if we can apply our filter locally (i.e.
         // with no server help) and we have the message we want to select.
         // Also update id_info accordingly.
         if (!filter.requires_adjustment_for_moved_with_target) {
             const superset_datasets = message_list_data_cache.get_superset_datasets(filter);
             for (const superset_data of superset_datasets) {
+                // Reset properties that might have been set.
+                id_info = Object.assign(id_info, original_id_info);
```